### PR TITLE
[MyPy] Minimal type checking on TIR schedule

### PR DESCRIPTION
**Pull request by mikepapadim**
This PR adds mypy type checking for schedule module.

The schedule module is properly typed, so this addition will prevent untyped type defs to be added in the future.

I added `type ignores` for objects returning through FFI and there were causing  `error: Module has no attribute`.

I would like to hear some feedback on how to tackle this kind of object.

### CODE REVIEW

**Comment by tqchen** [comment]
ignoring types from FFI is fine since we likely will always have a wrapper around FFI calls and we only need to type the wrappers

File: tests/scripts/task_mypy.sh [hunk 1]
```diff
@@ -18,8 +18,5 @@
 set +e
```
Comment (jcf94) [inline tests/scripts/task_mypy.sh:18]: Remove this?

Comment (mikepapadim) [reply tests/scripts/task_mypy.sh]: Done

**Review by jcf94 — APPROVED** [review]
LGTM. Thanks! @mikepapadim

**Pull request merged**